jdaviz/core/astrowidgets_api.py

```diff
@@ -173,10 +182,21 @@ def zoom_level(self):
         if self.shape is None:  # pragma: no cover
             raise ValueError('Viewer is still loading, try again later')
 
+        if hasattr(self, '_get_real_xy'):
+            if self.state.reference_data is not None:
+                image, i_ref = get_reference_image_data(self.jdaviz_app, self.reference)
+            else:
+                i_top = get_top_layer_index(self)
+                image = self.layers[i_top].layer

```

Comment:
The `else` does not make sense and probably will never be executed.

```suggestion
            image, i_ref = get_reference_image_data(self.jdaviz_app, self.reference)
            # TODO: Do we want top layer instead?
            # i_top = get_top_layer_index(self)
            # image = self.layers[i_top].layer
```